ir/src/main/scala/org/scalajs/core/ir/Printers.scala

```diff
@@ -1052,4 +1052,601 @@ object Printers {
     def complete(): Unit = ()
   }
 
+  abstract class RawIndentationManager extends IndentationManager {
+    private var indentSize = 0
+    private val indentString = "  "
+
+    override def indent(): Unit = indentSize += 1
+    override def undent(): Unit = indentSize -= 1
+
+    protected def print(s: String): Unit = {
+      out.write(indentString * indentSize)
+      out.write(s)
+    }
+
+    protected def println(s: String): Unit =
+      print(s + "\n")
+
+    override protected def println(): Unit =
+      out.write("\n")
+  }
+
+  class IRASTPrinter(protected val out: Writer) extends RawIndentationManager {
+    /** Pretty-prints the IR's AST nodes.
+      */
+    def printTopLevelTree(tree: Tree): Unit =
+      println(tree)
+
+    private final def println(tree: Tree): Unit = {
+      print(tree)
+      out.write("\n")
+    }
+
+    private final def printList(ts: List[Tree]): Unit = {
+      if (ts.isEmpty) {
+        print("List ()")
```

Comment:
In general, I think it would be best not to have a space before `(`. That would be more consistent with the coding style we use in the actual code.

This comment applies to all other occurrences of this pattern in this file.